src/main/java/com/dataloom/data/DataApi.java

```diff
@@ -23,19 +30,27 @@
     String GET_DATA_PATH = "getData";
 
     String SET_ID        = "setId";
-    String SYNC_ID        = "syncId";
+    String SYNC_ID       = "syncId";
     String SET_ID_PATH   = "{" + SET_ID + "}";
-    String SYNC_ID_PATH   = "{" + SYNC_ID + "}";
+    String SYNC_ID_PATH  = "{" + SYNC_ID + "}";
+
+    String FILE_TYPE     = "fileType";
 
     @GET( CONTROLLER + "/" + ENTITY_DATA + "/" + SET_ID_PATH )
-    Iterable<SetMultimap<FullQualifiedName, Object>> getEntitySetData( @Path( SET_ID ) UUID entitySetId );
+    Iterable<SetMultimap<FullQualifiedName, Object>> getEntitySetData(
+            @Path( SET_ID ) UUID entitySetId,
+            @Query( FILE_TYPE ) FileType fileType );
 
     // TODO Should discuss what this path should be
     @POST( CONTROLLER + "/" + ENTITY_DATA + "/" + SET_ID_PATH + "/" + GET_DATA_PATH )
     Iterable<SetMultimap<FullQualifiedName, Object>> getEntitySetData(
             @Path( SET_ID ) UUID entitySetId,
-            @Body GetEntitySetRequest req );
+            @Body GetEntitySetRequest req,
+            @Query( FILE_TYPE ) FileType fileType );
 
     @POST( CONTROLLER + "/" + ENTITY_DATA + "/" + SET_ID_PATH + "/" + SYNC_ID_PATH )
```

Comment:
It works out fine because UUID IIRC the DispatcherServlet is not aware of type information when making routing decisions. Thus when it is dispatching methods for routes /somepath/{pathVariable} and /somepath/constant it cannot tell the difference between the provided URIs and will preferentially route to pathVariable.